lib/modules/platform/bitbucket/schema.ts

```diff
@@ -0,0 +1,27 @@
+import { ZodSchema, z } from 'zod';
+import type { PagedResult, SourceResults } from './types';
+
+const BitbucketSourceTypeSchema = z.enum(['commit_directory', 'commit_file']);
+
+const SourceResultsSchema: ZodSchema<SourceResults> = z.object({
+  path: z.string(),
+  type: BitbucketSourceTypeSchema,
+  commit: z.object({
+    hash: z.string(),
+  }),
+});
+
```

Comment:
You can derive the type from the schema instead of creating an interface

```suggestion
const SourceResultsSchema = z.object({
  path: z.string(),
  type: BitbucketSourceTypeSchema,
  commit: z.object({
    hash: z.string(),
  }),
});

export type SourceResults = z.infer<typeof SourceResultsSchema>

```